### PR TITLE
fix: "Join Event" button experience for users without a ticket

### LINKED ISSUE #6819: Improve "Join Event" button experience for users without a ticket
Users are getting confused when they click "Join Event" and they receive an error pop up. They do not read it carefully. They think they already signed up for the event, even though they only signed up for the platform. Therefore please change the feature as follows:

* Change top right "Join Event" button to "Join Live Event"
* If the user does not have a ticket there is a pop up. Please change this as follows: If the user clicks "OK" on the pop up and is logged in, but does not have a ticket, direct the user to the "Tickets" section on the event page.

Screenshot from 2021-02-24 15-12-29

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #6819

#### Short description of what this resolves:
- Change top right "Join Event" button to "Join Live Event"
- If the user clicks "OK" on the pop up and is logged in, but does not have a ticket, direct the user to the "Tickets" section on the event page.

https://user-images.githubusercontent.com/61330148/109024914-3c3a3200-76e4-11eb-8f30-295d908d80a2.mp4

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/public/schedule.js [hunk 1]
```diff
@@ -2,4 +2,6 @@ import classic from 'ember-classic-decorator';
 import Controller from '@ember/controller';
 
 @classic
-export default class ScheduleController extends Controller {}
+export default class ScheduleController extends Controller {
+    preserveScrollPosition = true;
```
Comment (MeghalBisht) [inline app/controllers/public/schedule.js:6]: follow up #6791

**Comment by mariobehling** [comment]
Thank you. When the user is logged in but does not have a ticket yet, please scroll to the "Ticket" section.

**Comment by mariobehling** [comment]
When the user is on another page and clicks "Tickets" on the left-side menu, he/she gets directed to the tickets section of the event start page. The expected behavior is the same here.

File: app/controllers/public.js [hunk 2]
```diff
@@ -59,6 +65,19 @@ export default class PublicController extends Controller {
     this.activeSession = [];
   }
 
+  @action
+  scrollonJoinEvent() {
+    if (!this.authManager.currentUser) {
+      this.transitionToRoute('login');
+    } else {
+      const el = document.querySelector('#tickets');
```
Comment (iamareebjamal) [inline app/controllers/public.js]: Duplicate code, move to a function

Comment (MeghalBisht) [reply app/controllers/public.js]: Pls check the new commit.

**Comment by mariobehling** [comment]
If the user clicks "ok" after the message the URL has always has "?video_dialog=true" appended. Is that necessary?

If the user is on another page e.g. https://open-event-frontend-git-fork-meghalbisht-joinevent-eventyay.vercel.app/e/fa96ae2c/schedule and clicks "Join Live Event" the link to the tickets section does not work as expected. Ember does not make it easy to link to sections in pages but there is a work around. Please check out how this works.

**Comment by MeghalBisht** [comment]
**Update -** 
https://user-images.githubusercontent.com/61330148/109431206-526d2880-7a2b-11eb-9bb5-135a72425508.mp4

**Review by mariobehling — APPROVED** [review]
Thank you!

**Pull request merged**